pkg/fleet/installer/service/apm_inject.go

```diff
@@ -125,6 +125,10 @@ func (a *apmInjectorInstaller) Setup(ctx context.Context) error {
 	if err := a.configureSocketsEnv(ctx); err != nil {
 		return err
 	}
+	if err := os.Symlink(envFilePath, envFilePathOld); err != nil && !os.IsExist(err) {
+		// TODO(baptiste): remove this once the injector has been successfully migrated
+		return fmt.Errorf("failed to symlink %s to /var/run/datadog-installer/environment: %w", envFilePath, err)
```

Comment:
it's ok to fail in this case, especially if /var/run has been removed

Reply:
what about moving all to the installer: symlink the directories run/datadog-installer, like this the logic sits only in one place

Reply:
done in [59349f2](https://github.com/DataDog/datadog-agent/pull/28537/commits/59349f2a8c5e77a30a84b1df915ba3c13d66d333)

Reply:
> what about moving all to the installer: symlink the directories run/datadog-installer, like this the logic sits only in one place

That works too. I'll add it.